pkg/iris/eventitems.go

```diff
@@ -0,0 +1,23 @@
+package iris
+
+type DiscoveryItem struct {
+	ID      string      `json:"id"`
+	Scope   string      `json:"scope"`
+	Type    string      `json:"type"`
+	Source  string      `json:"source"`
+	Time    string      `json:"time"`
+	Subject string      `json:"subject"`
+	Data    interface{} `json:"data"`
+}
+
+func GenerateDiscoveryItem(id string, scope string, t string, source string, time string, subject string, data interface{}) *DiscoveryItem {
```

Comment:
If it is some sort of factory pattern then the Prefix should consistent across the project. e.q. "NewDiscoveryItem" instead of "Generate..."

Also one more critical question; what is the benefit of having a Factory function if it simply create the struct but does not encapsulate any more logic? Couldn't we then just just the DiscoveryItem{...} call directly ?

Reply:
makes sense ; will do the changes